activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -141,6 +141,10 @@ def supports_datetime_with_precision?
         end
       end
 
+      def supports_virtual_columns?
+        !mariadb? && version >= "5.7.5"
```

Comment:
I do not have MariaDB environment yet, looks like it always returns false with MariaDB.  MariaDB 5.2 supports virtual columns and this pull request is considering it by implementing PERSISTENT option.  Forgive me if I misread how ! works...